administrator/components/com_workflow/Model/StageModel.php

```diff
@@ -292,36 +293,132 @@ public function setDefault($pk, $value = 1)
 	}
 
 	/**
-	 * Method to change the published state of one or more records.
+	 * Method to change the published state of one or more stages.
 	 *
 	 * @param   array    &$pks   A list of the primary keys to change.
-	 * @param   integer  $value  The value of the published state.
+	 * @param   integer  $value     The target condition
 	 *
 	 * @return  boolean  True on success.
 	 *
 	 * @since  __DEPLOY_VERSION__
 	 */
-	public function publish(&$pks, $value = 1)
+	public function publish(&$pks, $value = ContentComponent::CONDITION_PUBLISHED)
 	{
-		$table = $this->getTable();
+		$db  = $this->getDbo();
 		$pks   = (array) $pks;
 		$app = Factory::getApplication();
-		$extension = $app->getUserStateFromRequest('com_workflow.state.filter.extension', 'extension', null, 'cmd');
 
-		// Default item existence checks.
-		if ($value != 1)
+		$return = true;
+
+		if ($value != (int) ContentComponent::CONDITION_PUBLISHED)
+		{
+			// Clear pks 
+			self::checkDefaultStage($pks, $value);
+		}
+
+		if (!empty($pks))
+		{
+			// Change published for all stages
+			$return =  parent::publish($pks, $value);
+
+			// If the stage is trashed, the transitions to and from this stage must be trashed too
+			if ($return && $value == (int) ContentComponent::CONDITION_TRASHED)
+			{
+				$db = $this->getDbo();
+				$query = $db->getQuery(true)
+					->update($db->quoteName('#__workflow_transitions'))
+					->set($db->quoteName('published') . ' = ' . ContentComponent::CONDITION_TRASHED)
+					->where($db->quoteName('from_stage_id') . ' IN (' . implode(',', $pks) . ') OR'  
+						. $db->quoteName('to_stage_id') . ' IN (' . implode(',', $pks) . ')' );
+
+				$db->setQuery($query)->execute();
+			}
+		}
+
+		return $return;
+	}
+	
+	/**
+	 * Method to delete one or more stages.
+	 *
+	 * @param   array    &$pks   A list of the primary keys to delete.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public function delete(&$pks)
+	{
+		$db  = $this->getDbo();
+		$pks   = (array) $pks;
+		$app = Factory::getApplication();
+
+		// Clear pks
+		self::checkDefaultStage($pks, ContentComponent::CONDITION_TRASHED);
+
+		if (!empty($pks))
 		{
-			foreach ($pks as $i => $pk)
+			// Change published for all stages where it is possible
+			$return =  parent::delete($pks);
+		}
+
+		if ($return)
+		{
+			// Delete the transitions to and from deleted stages
+			$query = $db->getQuery(true)
+				->delete($db->quoteName('#__workflow_transitions'))
+				->where($db->quoteName('to_stage_id') . ' = ' . (int) $pk, 'OR')
+				->whereIn($db->quoteName('from_stage_id'), $pks);
```

Comment:
Das stimmt glaub nicht, das ```(int) $pk``` müsste wahrscheinlich auch ein IN(...) sein?

Reply:
done